libs/@blockprotocol/graph/src/codegen/postprocess/generate-link-and-target-definitions.ts

```diff
@@ -0,0 +1,237 @@
+import { VersionedUrl } from "@blockprotocol/type-system/slim";
+
+import { mustBeDefined } from "../../shared/util/must-be-defined.js";
+import {
+  typedEntries,
+  typedValues,
+} from "../../shared/util/typed-object-iter.js";
+import { PostprocessContext } from "../context/postprocess.js";
+import { entityDefinitionNameForEntityType } from "../shared";
+
+type IdentifierToCompiledDefinition = {
+  identifier: string;
+  compiledContents: string;
+  dependentOnIdentifiers: string[];
+};
+
+/**
+ * Create the individual definition of a `LinkAndTargetEntity` for a given outgoing link from a source
+ *
+ * @param sourceName
+ * @param linkIdentifier
+ * @param targetIdentifiers
+ */
+const individualOutgoingLinkAndTargetDefinition = (
+  sourceName: string,
+  linkIdentifier: string,
+  targetIdentifiers: [string, ...string[]],
+): IdentifierToCompiledDefinition => {
+  const identifier = `${sourceName}${linkIdentifier}Links`;
+
+  const targetUnion = targetIdentifiers.join(" | ");
+  const compiledContents = `export type ${identifier} = { linkEntity: ${linkIdentifier}; target: ${targetUnion} }`;
```

Comment:
Here we've encoded the `rightEntity` as the `target`, and this type is therefore inconsistent with the typings of a subgraph. 

I think this should be `rightEntity`, unless we want to change `rightEntity` to `target` everywhere else

Reply:
So it is, I changed these about 5 times trying to figure out which one was consistent, clearly I missed something in the refactoring. (Yet again I really wish we could use `satisfies` on types to describe a subtype)

Applied as driveby in
[Rename OutgoingLinksAndTargets union to OutgoingLinkAndTarget](https://github.com/blockprotocol/blockprotocol/pull/1158/commits/9f0576eb32a34230090296de4daf0fa0d82c6625)